From 747df57eabbe2e3b5bd47e268aa42658a57749ca Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Wed, 3 Feb 2021 16:23:43 +0000
Subject: [PATCH] ARM: dts: Declare Pi400 and CM4 have no audio pins

The audio_pins node is left as a placeholder for the audremap overlay,
and it must have (empty) brcm,function and brcm,pins properties
otherwise it will be rejected by the pinctrl driver.

See: https://www.raspberrypi.org/forums/viewtopic.php?f=98&t=301891

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/bcm2711-rpi-400.dts | 4 ++--
 arch/arm/boot/dts/bcm2711-rpi-cm4.dts | 2 ++
 2 files changed, 4 insertions(+), 2 deletions(-)

--- a/arch/arm/boot/dts/bcm2711-rpi-400.dts
+++ b/arch/arm/boot/dts/bcm2711-rpi-400.dts
@@ -495,8 +495,8 @@
 
 &gpio {
 	audio_pins: audio_pins {
-		brcm,pins = <40 41>;
-		brcm,function = <4>;
+		brcm,pins = <>;
+		brcm,function = <>;
 	};
 };
 
--- a/arch/arm/boot/dts/bcm2711-rpi-cm4.dts
+++ b/arch/arm/boot/dts/bcm2711-rpi-cm4.dts
@@ -434,6 +434,8 @@
 
 &gpio {
 	audio_pins: audio_pins {
+		brcm,pins = <>;
+		brcm,function = <>;
 	};
 };
 
